### PR TITLE
Add support for Google Authentication.

### PR DESCRIPTION
**Pull request by fidalgo**
I've implemented support for Google Authentication, please review and let me know your thoughts.
In order to support both Facebook and Google a new column 'provider' was added.
The 'facebook_uid' column was renamed to uid and represent an Facebook uid or Google sud.

### CODE REVIEW

**Comment by oliverbarnes** [comment]
Thanks for this @fidalgo, looking good. Would you also add a request spec like [this one for facebook auth](https://github.com/simplabs/rails_api_auth/blob/master/spec/requests/oauth2_spec.rb#L41), please?

Coincidentally, I am about to [extract an identity model](https://github.com/simplabs/rails_api_auth/issues/15) to make adding new identity providers easier

**Comment by marcoow** [comment]
Would be cool to extract a Base service that both Facebook and Google extend so we have less duplication. Also this breaks the tests.

**Comment by fidalgo** [comment]
@oliverbarnes I missed that one but I will do it for sure.
@marcoow That's on my TODO list to extract a base service and keep things more OO.

**Comment by marcoow** [comment]
@fidalgo: would be great to merge this with the base service included 👍

**Comment by fidalgo** [comment]
So now I believe it's ready for another look!

File: app/services/facebook_authenticator.rb [hunk 1]
```diff
@@ -3,79 +3,47 @@
...
-      connect_login_to_fb_account(login, facebook_user)
-    else
-      login = create_login_from_fb_account(facebook_user)
-    end
-
-    login
-  end
+  PROVIDER = 'facebook'
+  TOKEN_URL = 'https://graph.facebook.com/oauth/access_token?client_id=%{client_id}&client_secret=%{client_secret}&code=%{auth_code}&redirect_uri=%{redirect_uri}'
+  PROFILE_URL = 'https://graph.facebook.com/me?fields=email,name&access_token=%{access_token}'
```
Comment (marcoow) [inline app/services/facebook_authenticator.rb]: The graph URL should be read from the config (`RailsApiAuth.facebook_graph_url`).

File: app/services/google_authenticator.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+require 'httparty'
+
+# Handles Google authentication
+#
+# @!visibility private
+class GoogleAuthenticator < BaseAuthenticator
+
+  PROVIDER = 'google'
+  TOKEN_URL = 'https://www.googleapis.com/oauth2/v3/token'
+  PROFILE_URL = 'https://www.googleapis.com/plus/v1/people/me/openIdConnect?access_token=%{access_token}'
```
Comment (marcoow) [inline app/services/google_authenticator.rb]: The API URL should be stored in the config as well (see above).

**Comment by marcoow** [comment]
Thanks @fidalgo, looks great! Just 2 small things: we should read the API URLs for Facebook and Google from the config so people can configure the API versions they want to use.

Looking forward to merge this!

**Comment by fidalgo** [comment]
@marcoow Well I have a different opinion, because we are not exposing those API's to other apps, so it will avoid people setting API versions we do not support and then breaking the gem functionality.
Imagine the case when in another API version `email` or `sud` in Google or `id` in Facebook parameters get renamed, we'll get a bunch of errors if the user sets that version URL.

Of course I could be missing something...

**Comment by marcoow** [comment]
@fidalgo: hm, that's a good point. I guess we should remove the setting then though. Also, can you make sure we use the latest versions of each API (if not already the case)?

**Comment by fidalgo** [comment]
@marcoow The Google Authentication is using the latest API versioin, will check for Facebook.

**Comment by fidalgo** [comment]
@marcoow I have updated the code to comply with the 2.4 version of Facebook API (the latest).

**Comment by marcoow** [comment]
The `facebook_graph_url` setting ca be removed then.

**Comment by fidalgo** [comment]
@marcoow That setting was already removed, but now I've updated the Readme to reflect the change.

**Comment by marcoow** [comment]
@fidalgo 👍thanks a lot!

**Pull request merged**